pkg/broker/filter/filter_handler.go

```diff
@@ -203,6 +211,38 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 		return
 	}
 
+	features := feature.FromContext(ctx)
+	if features.IsOIDCAuthentication() {
+		h.logger.Debug("OIDC authentication is enabled")
+		subs, err := h.getSubscription(t)
+		if err != nil {
+			h.logger.Info("Unable to get the subscription", zap.Error(err), zap.Any("triggerRef", triggerRef))
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		if subs.Spec.Subscriber.Audience == nil {
+			h.logger.Warn(fmt.Sprintf("Audience of subscription %s/%s must not be nil, while feature %s is enabled", subs.Name, subs.Namespace, feature.OIDCAuthentication))
+			writer.WriteHeader(http.StatusInternalServerError)
+			return
+		}
```

Comment:
Actually we don't need to check the Subscription here. For requests to the mt-broker-filter, we have the "hard coded" `mt-broker-filter` audience as a contract. So each request to the broker-filter should use this audience.

When we create the subscription of a trigger, we set the audience of the destination to `mt-broker-filter`:
https://github.com/knative/eventing/blob/223846c4f81935409f147706b21045ce00cadad2/pkg/reconciler/broker/trigger/trigger.go#L229-L231

https://github.com/knative/eventing/blob/223846c4f81935409f147706b21045ce00cadad2/pkg/reconciler/broker/trigger/trigger.go#L66


I think this then makes it easier, as you don't need the subscription lister, etc.